application/forms/Command/Object/ScheduleServiceDowntimeForm.php

```diff
@@ -106,14 +156,16 @@ protected function assembleElements()
         );
         $decorator->decorate($this->getElement('flexible'));
 
-        if ($this->getPopulatedValue('flexible') === 'y') {
+        if ($isFlexible) {
+            $this->getElement('start')->setValue($this->start);
+            $this->getElement('end')->setValue($this->flexibleEnd);
```

Comment:
Why? Isn't this already done in the element definition?

Reply:
I did this because when I set the type back to `fixed`, the value did not change back to `fixed_downtime_end`